src/ngamsUtils/ngamsUtils/ngasXSyncTool.py

```diff
@@ -0,0 +1,1326 @@
+    """
+    Get the list of ready NAUs in a cluster
+
+    :param connection: DB connection (ngamsDb)
+    :param target_cluster: Target cluster name (string)
+    :return: List of NAUs (list)
+    """
+    sql = "select host_id, srv_port from ngas_hosts " \
+          "where cluster_name = '{:s}' and host_id in " \
+          "(select host_id from ngas_disks where completed = 0 and mounted = 1) order by host_id"
+    sql = sql.format(target_cluster)
+    result = connection.query2(sql)
+    if result == [[]]:
+        return []
+    else:
+        host_list = []
+        for node in result[0]:
+            host_list.append("{:s}:{:s}".format(node[0], node[1]))
+        return host_list
+
+
+def get_cluster_nodes(connection, target_cluster):
+    """
+    Get the list of nodes in the cluster
+
+    :param connection: DB connection (ngamsDb)
+    :param target_cluster: Target cluster name (string)
+    :return: Tuple with three elements:
+                1. List containing node:port pairs
+                2. String buffer with comma separated list of hostnames
+                3. String buffer with a comma separated list of host:port pairs (tuple)
+    """
+    sql = "select host_id, srv_port from ngas_hosts where cluster_name = '{:s}'".format(target_cluster)
+    result = connection.query2(sql)
+    if result == [[]]:
+        return []
+    else:
+        host_list = []
+        host_list_str = ""
+        server_list = ""
+        for node in result[0]:
+            host_list.append("{:s}:{:s}".format(node[0], node[1]))
+            host_list_str += "'{:s}',".format(node[0])
+            server_list += "{:s}:{:s},".format(node[0], node[1])
+        return host_list, host_list_str[:-1], server_list[:-1]
+
+
+def check_if_file_in_target_cluster(connection, cluster_nodes, file_id, file_version):
+    """
+    Query the information for a file
+
+    :param connection: DB connection (ngamsDb)
+    :param cluster_nodes: List of nodes in the target cluster (list)
+    :param file_id: ID of file to check for (string)
+    :param file_version: Version of file (integer)
+    :return: Information for file as tuple:
+                (<Disk Id>, <File ID>, <File Version>) or None if the file is not available in the cluster (tuple|None)
+    """
+    logger.info("Entering check_if_file_in_target_cluster() ...")
+
+    # Reformat the cluster node list to a string for passing into the SQL statement
+    # It should be in the format "'ngas04:7777', 'ngas04:7778'"
+    cluster_nodes_sql = str(cluster_nodes.split(',')).replace('[', '').replace(']', '')
+
+    sql = "select nf.disk_id, nf.file_id, nf.file_version " \
+          "from ngas_files nf, ngas_disks nd " \
+          "where nf.file_id = '{:s}' and nf.file_version = {:d} and nf.disk_id = nd.disk_id " \
+          "and (nd.host_id in ({:s}) or nd.last_host_id in ({:s}))"
+    sql = sql.format(file_id, file_version, cluster_nodes_sql, cluster_nodes_sql)
+    result = connection.query2(sql)
```

Comment:
One of the advantages of the newer `query2` over the old `query` is that it receives the SQL query parameters as arguments and internally binds them to prepared SQL statements, so users don't have to format SQL statements manually. So in this case you'll want to use something like:

```
sql = ("select nf.disk_id, nf.file_id, nf.file_version " +
       "from ngas_files nf, ngas_disks nd " +
       "where nf.file_id = {0} and nf.file_version = {1} and nf.disk_id = nd.disk_id " +
       "and (nd.host_id in {2} or nd.last_host_id in {3})")
result = connection.query2(sql, args=(file_id, file_version, cluster_nodes_sql, cluster_nodes_sql))
```

That should work, I think. If it doesn't then please let's iterate over it.

Reply:
I will follow your suggestion